Classes/Emogrifier.php

```diff
@@ -1767,6 +1793,18 @@ public function handleXpathError($type, $message, $file, $line, array $context)
             );
         }
 
+        // Catches eventual warnings generated by method getAllNodesWithStyleAttribute()
+        if ($type === E_WARNING && isset($context['xPath'])) {
+            throw new \RuntimeException(
+                sprintf(
+                    '%s in %s on line %s',
+                    $message,
+                    $file,
+                    $line
+                )
+            );
```

Comment:
While you're at it, could you please add unique exception codes to this exception and the one above?

And: will the `sprintf` parameters fit in one line, and could you please use numbered parameters?

Reply:
`sprintf`: Do you mean using `%u` for the exception code or `%1$s` and `%2$s` (as in argument swapping)?

Reply:
Actually, both (numbered arguments and typed arguments).